packages/features/orgs/SubdomainProvider.tsx

```diff
@@ -0,0 +1,64 @@
+import { useRouter } from "next/router";
+import type { ReactNode } from "react";
+import { createContext, useContext, useState, useEffect } from "react";
+
+import { trpc } from "@calcom/trpc";
+
+const useSubdomainCheck = () => {
+  const router = useRouter();
+  const [isSubdomain, setIsSubdomain] = useState(false);
+  const [subdomain, setSubdomain] = useState<string | null>(null);
+
+  useEffect(() => {
+    const checkSubdomain = () => {
+      const url = new URL(router.asPath, window.location.origin);
+      const hostnameParts = url.hostname.split(".");
+
+      if (hostnameParts.length >= 3) {
+        const subdomain = hostnameParts.slice(0, -2).join(".");
+        const domain = hostnameParts.slice(-2).join(".");
+        if (domain === "cal.com" || domain === "cal.dev") {
+          if (subdomain !== "app" && subdomain !== "console") {
+            setIsSubdomain(true);
+            setSubdomain(subdomain);
+          }
+        }
+      }
+    };
+    if (router.isReady) {
+      checkSubdomain();
+    }
+  }, [router]);
```

Comment:
Please test - this shouldnt cause a re-render in the while tree as react is kinda smart and realises the values of the provider dont change. 

We should only get one render per navigation still.

Reply:
Can we get this from the server side?